AutoscoperM/AutoscoperM.py

```diff
@@ -1194,3 +1281,62 @@ def convertNodeToData(self, volumeNode: slicer.vtkMRMLVolumeNode) -> vtk.vtkImag
             imageData = imageReslice.GetOutput()
 
         return imageData
+
+    def check4D(self, node: slicer.vtkMRMLNode) -> bool:
+        """
+        Checks if the volume is 4D
+        """
+        self.is_4d = type(node) == slicer.vtkMRMLSequenceNode
+
+    def getItemInSequence(self, sequenceNode: slicer.vtkMRMLSequenceNode, idx: int) -> slicer.vtkMRMLNode:
+        """
+        Returns the item at the specified index in the sequence node
+
+        :param sequenceNode: sequence node
+        :param idx: index
+
+        :return: item at the specified index
+        """
+        if type(sequenceNode) != slicer.vtkMRMLSequenceNode:
+            logging.error("[AutoscoperM.logic.getItemInSequence] sequenceNode must be a sequence node")
+            return None
+
+        if idx >= sequenceNode.GetNumberOfDataNodes():
+            logging.error(f"[AutoscoperM.logic.getItemInSequence] index {idx} is out of range")
+            return None
+
+        browserNode = slicer.modules.sequences.logic().GetFirstBrowserNodeForSequenceNode(sequenceNode)
+        browserNode.SetSelectedItemNumber(idx)
+        return browserNode.GetProxyNode(sequenceNode), sequenceNode.GetNthDataNode(idx).GetName()
+
+    def getNextItemInSequence(self, sequenceNode: slicer.vtkMRMLSequenceNode) -> slicer.vtkMRMLNode:
+        """
+        Returns the next item in the sequence
+
+        :param sequenceNode: sequence node
+
+        :return: next item in the sequence
+        """
+        if type(sequenceNode) != slicer.vtkMRMLSequenceNode:
+            logging.error("[AutoscoperM.logic.getNextItemInSequence] sequenceNode must be a sequence node")
+            return None
+
+        browserNode = slicer.modules.sequences.logic().GetFirstBrowserNodeForSequenceNode(sequenceNode)
+        browserNode.SelectNextItem()
+        idx = browserNode.GetSelectedItemNumber()
+        return browserNode.GetProxyNode(sequenceNode), sequenceNode.GetNthDataNode(idx).GetName()
+
+    def cleanFilename(self, volumeName: str, index: Optional[int] = None) -> str:
```

Comment:
Consider first calling `slicer.qSlicerCoreIOManager.forceFileNameValidCharacters` and then performing only the additional steps specific to Autoscoper